cmd/main.go

```diff
@@ -295,6 +324,78 @@ func runWebhook(args []string) {
 	}
 }
 
+func runMutate(args []string) {
+	fs := flag.NewFlagSet("mutate", flag.ExitOnError)
+	var mutateFlags MutateFlags
+	mutateFlags.AddFlags(fs)
+
+	parseFlagsOrDie(fs, args)
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(mutateFlags.ZapOptions)))
+
+	// Validate required flags
+	if mutateFlags.PipelineRunFile == "" {
+		fmt.Fprintf(os.Stderr, "Error: --pipelinerun-file is required\n")
+		fs.Usage()
+		os.Exit(1)
+	}
+	if mutateFlags.ConfigDir == "" {
+		fmt.Fprintf(os.Stderr, "Error: --config-dir is required\n")
+		fs.Usage()
+		os.Exit(1)
+	}
+
+	// Load PipelineRun from file
+	pipelineRunData, err := os.ReadFile(mutateFlags.PipelineRunFile)
+	if err != nil {
+		setupLog.Error(err, "Failed to read PipelineRun file", "file", mutateFlags.PipelineRunFile)
+		os.Exit(1)
+	}
+
+	var pipelineRun tekv1.PipelineRun
+	if err := yaml.Unmarshal(pipelineRunData, &pipelineRun); err != nil {
+		setupLog.Error(err, "Failed to parse PipelineRun YAML", "file", mutateFlags.PipelineRunFile)
+		os.Exit(1)
+	}
+
+	// Load config
+	cfg, err := loadConfig(mutateFlags.ConfigDir)
+	if err != nil {
+		setupLog.Error(err, "unable to load configuration")
+		os.Exit(1)
+	}
+
+	// Compile CEL programs and create mutator
+	programs, err := cel.CompileCELPrograms(cfg.CEL.Expressions)
+	if err != nil {
+		setupLog.Error(err, "unable to compile CEL programs")
+		os.Exit(1)
+	}
+	mutator := cel.NewCELMutator(programs)
+
+	// Create custom defaulter
+	customDefaulter, err := webhookv1.NewCustomDefaulter(cfg.QueueName, []webhookv1.PipelineRunMutator{mutator})
+	if err != nil {
+		setupLog.Error(err, "Unable to create custom defaulter")
+		os.Exit(1)
+	}
+
+	// Apply mutation
+	ctx := context.Background()
```

Comment:
we should handle Signals from the OS so that the user can Ctrl+C the tool. Something like:

```suggestion
	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
	defer cancel()
```